src/nixpacks/cache.rs

```diff
@@ -1,5 +1,16 @@
 pub fn sanitize_cache_key(cache_key: String) -> String {
-    cache_key.replace(' ', "-")
+    cache_key
+        .chars()
+        .filter(|x| match x {
+            // remove these chars
+            '/' | '.' => false,
+            _ => true,
+        })
+        .map(|x| match x {
+            ' ' | '~' => '-',
```

Comment:
We should be able to safely map `~` to `/root`. That would allow cache directories to be something like `~/.cache/something`, which is fairly common in the unix world. At the moment users have nixpacks need to know that root is `/root`

Reply:
mapping `~` to `/root` I think should be for the target cache directory 
I've implemented & pushed please have a look, this is a sample result for one of my local runs

`CACHED [stage-0 6/8] RUN --mount=type=cache,id=DdFtDTR8ns0--root-m2,target=/root/.m2 `